### PR TITLE
VMS: don't use app_malloc() in apps/lib/vms_decc_argv.c

### PR DESCRIPTION
**Pull request by levitte**
The reason being that it would otherwise force test programs to link
with all of libapps.a, which unfortunately causes multiple symbol
definition issues.

The quick and dirty fix is to use malloc() instead of app_malloc() in
apps/lib/vms_decc_argv.c, and clean up libapps.a later.

-----

This is a quick'n'dirty alternative to #15342

### CODE REVIEW

**Comment by levitte** [comment]
I'm keeping this in draft until I've verified that it works as expected

**Comment by slontis** [comment]
Is there a reason not to just use OPENSSL_malloc()?  All the frees would be done with OPENSSL_free().

**Comment by levitte** [comment]
> Is there a reason not to just use OPENSSL_malloc()? All the frees would be done with OPENSSL_free().

I..  haven't tried that, didn't even think of it.  Will do

**Comment by levitte** [comment]
This is now confirmed to build correctly

**Comment by slontis** [comment]
I dont see a problem with this approach.. I would not even consider it hacky now.

File: apps/lib/vms_decc_argv.c [hunk 1]
```diff
@@ -51,7 +50,12 @@ char **copy_argv(int *argc, char *argv[])
 
     cleanup_argv();
 
-    newargv = app_malloc(sizeof(*newargv) * (count + 1), "argv copy");
+    /*
+     * We purposefully use malloc() rather than app_malloc() here, to avoid
```
Comment (paulidale) [inline apps/lib/vms_decc_argv.c:54]: The comment is now wrong.

**Review by paulidale — CHANGES_REQUESTED** [review]

File: apps/lib/vms_decc_argv.c (hunk 1 above)
Comment (levitte) [reply apps/lib/vms_decc_argv.c]: Fixed

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Much nicer approach.

**Comment by levitte** [comment]
> Much nicer approach.

Yeah well, libapps is still of mess...  but at least, we don't need to rush clearing that up

**Comment by paulidale** [comment]
There is quite a bit of cleanup I'd like to do in apps....

**Comment by levitte** [comment]
> There is quite a bit of cleanup I'd like to do in apps....

[ahem] I'll raise you one...  I'd like to rebuild it all from scratch, and as a separate project.

**Comment by levitte** [comment]
Merged

a066841554bd23281ae4bb48badc088753f734ca VMS: don't use app_malloc() in apps/lib/vms_decc_argv.c

**Pull request closed without merging**